### PR TITLE
WIP Issue1369, resolving parameters in top module

### PR DESCRIPTION
**Pull request by emacdo12**
#### Description

Added function that'll resolve the parameters that are used to describe overrides of parameters in other instances.

#### Related Issue

Issue1369 - main repository

#### Types of changes

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by emacdo12** [comment]
I think it's ready to merge, I can make it into one commit then put it on the main repo when you look it over

File: ODIN_II/SRC/netlist_create_from_ast.cpp [hunk 1]
```diff
@@ -5143,4 +5147,81 @@ void reorder_connections_from_name(ast_node_t* instance_node_list, ast_node_t* i
...
+        } else {
+            ast_node_t* newNode = ast_node_deep_copy((ast_node_t*)local_param_table_sc->data[sc_spot]);
+            if (newNode->type != NUMBERS) {
+                newNode = resolve_top_parameters_defined_by_parameters(newNode, top_sc_list, count);
+            }
+            node = free_whole_tree(node);
+            node = newNode;
+        }
+    }
+    if (node->type == BINARY_OPERATION) {
```
Comment (jeanlego) [inline ODIN_II/SRC/netlist_create_from_ast.cpp:5207]: Don't pre declare your variable for your for loop, this was necessary in old C (C89) but now it is not necessary and I would dare say bad pratice.

File: ODIN_II/SRC/netlist_create_from_ast.cpp [hunk 2]
```diff
@@ -5143,4 +5147,81 @@ void reorder_connections_from_name(ast_node_t* instance_node_list, ast_node_t* i
...
+        }
+        node->children[i] = new_child;
+        i++;
+    }
+    return (node);
+}
+
+ast_node_t* resolve_top_parameters_defined_by_parameters(ast_node_t* node, sc_hierarchy* top_sc_list, int count) {
+    int i;
+    const char* string;
```
Comment (jeanlego) [inline ODIN_II/SRC/netlist_create_from_ast.cpp]: Make sure you initialize all your variable, i.e. set pointers to NULL

**Comment by jeanlego** [comment]
I think you are close to done here if you just do a few tweak and its good to go :)

**Comment by jeanlego** [comment]
You can move this to the main repo! :) Make sure to fix the format otherwise Travis throws an error.

**Comment by jeanlego** [comment]
this was merged so closing :) good job

**Pull request closed without merging**